src/sonic-host-services/scripts/hostcfgd

```diff
@@ -897,145 +910,179 @@ class HostConfigDaemon:
+        self.subscribe('RADIUS', lambda table, key, op, data: self.radius_global_handler(key, op, data), PRIO_INIT-2)
+        self.subscribe('RADIUS_SERVER', lambda table, key, op, data: self.radius_server_handler(key, op, data), PRIO_INIT-2)
+        # Handle IPTables configuration
+        self.subscribe('LOOPBACK_INTERFACE', lambda table, key, op, data: self.lpbk_handler(key, op, data), PRIO_INIT-3)
+        # Handle NTP & NTP_SERVER updates
+        self.subscribe('NTP', lambda table, key, op, data: self.ntp_global_handler(key, op, data), PRIO_INIT-4)
+        self.subscribe('NTP_SERVER', lambda table, key, op, data: self.ntp_server_handler(key, op, data), PRIO_INIT-4)
+        # Handle updates to src intf changes in radius
+        self.subscribe('MGMT_INTERFACE', lambda table, key, op, data: self.mgmt_intf_handler(key, op, data), PRIO_INIT-5)
+        self.subscribe('VLAN_INTERFACE', lambda table, key, op, data: self.vlan_intf_handler(key, op, data), PRIO_INIT-5)
+        self.subscribe('VLAN_SUB_INTERFACE', lambda table, key, op, data: self.vlan_sub_intf_handler(key, op, data), PRIO_INIT-5)
+        self.subscribe('PORTCHANNEL_INTERFACE', lambda table, key, op, data: self.portchannel_intf_handler(key, op, data), PRIO_INIT-5)
+        self.subscribe('INTERFACE', lambda table, key, op, data: self.phy_intf_handler(key, op, data), PRIO_INIT-5)
+        
         syslog.syslog(syslog.LOG_INFO,
                       "Waiting for systemctl to finish initialization")
         self.wait_till_system_init_done()
         syslog.syslog(syslog.LOG_INFO,
                       "systemctl has finished initialization -- proceeding ...")
 
-        # Update all feature states once upon starting
-        self.feature_handler.update_all_features_config()
-
-        # Defer load until subscribe
-        self.load()
-
-        self.config_db.listen()
+    def start(self):
+        init_load_flag = True
+        while True:
+            state, selectable_ = self.selector.select(SELECT_TIMEOUT)
+            if state == self.selector.TIMEOUT:
+                continue
+            elif state == self.selector.ERROR:
+                raise Exception("Received error from select")
+
+            fd = selectable_.getFd()
+            # Get the Corresponding subscriber & table
+            subscriber, table = self.subscriber_map.get(fd, (None, ""))
+            if not subscriber:
+                syslog.syslog(syslog.LOG_ERR,
+                        "No Subscriber object found for fd: {}, subscriber map: {}".format(fd. subscriber_map))
+                continue
+            key, op, fvs = subscriber.pop()
+            # Get the registered callback
+            cbs = self.callbacks.get(table, None)
+            for callback in cbs:
+                callback(table, key, op, dict(fvs))
+            
+            if init_load_flag and table == "FEATURE" and key.lower() == "swss":
```

Comment:
What is the purpose of this condition?

Reply:
Previously, before the listen phase was started, `update_all_features_config` was run followed by `load`  previously. `load` calls the `load` method of AaaCfg. 

With the redesign, we cannot have the ordering which we had previously, thus introduced this condition i.e. to basically execute the `load ` of aaacfg after swss is started

Reply:
Why AAA config depends on FEATURE config?

Reply:
AAA/RADIUS/TACACS tables, have src_intf option which could be configured as either eth0, Loopback0 etc. 

https://github.com/Azure/SONiC/blob/master/doc/aaa/radius_authentication.md#radius-table-schema

So, i thought of having the SWSS configured first and then AAA. but if you think it's safe to confiure AAA before feature config, then i can change this. 

Alteast previously, that is not the case, AAA config was run after the FEATURE config is completed

Reply:
You're saying that AAA/RADIUS/TACACS need to have an interface to work with, which could be management interface or loopback interface which is created by swss and this is why swss has to be started first and then AAA configuration processed?
If it is, I would suggest removing this condition, since it anyway does not guarantee that Loopback interface will be created at this time.

Reply:
Done